In [2]:
import pandas as pd
from google.colab import drive

In [3]:
# Mount Google Drive
drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


In [5]:
# Load dataset
dataset_path = "/content/drive/MyDrive/SEM6/Prak. Pembelajaran Mesin/materi3/student_depression_dataset.csv"
df = pd.read_csv(dataset_path)
df.head()

id  Gender   Age           City Profession  Academic Pressure  \
0   2    Male  33.0  Visakhapatnam    Student                5.0   
1   8  Female  24.0      Bangalore    Student                2.0   
2  26    Male  31.0       Srinagar    Student                3.0   
3  30  Female  28.0       Varanasi    Student                3.0   
4  32  Female  25.0         Jaipur    Student                4.0   

   Work Pressure  CGPA  Study Satisfaction  Job Satisfaction  \
0            0.0  8.97                 2.0               0.0   
1            0.0  5.90                 5.0               0.0   
2            0.0  7.03                 5.0               0.0   
3            0.0  5.59                 2.0               0.0   
4            0.0  8.13                 3.0               0.0   

        Sleep Duration Dietary Habits   Degree  \
0          '5-6 hours'        Healthy  B.Pharm   
1          '5-6 hours'       Moderate      BSc   
2  'Less than 5 hours'        Healthy       BA   
3          '7-8 hours'       Moderate      BCA   
4          '5-6 hours'       Moderate   M.Tech   

  Have you ever had suicidal thoughts ?  Work/Study Hours Financial Stress  \
0                                   Yes               3.0              1.0   
1                                    No               3.0              2.0   
2                                    No               9.0              1.0   
3                                   Yes               4.0              5.0   
4                                   Yes               1.0              1.0   

  Family History of Mental Illness  Depression  
0                               No           1  
1                              Yes           0  
2                              Yes           0  
3                              Yes           1  
4                               No           0

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27901 entries, 0 to 27900
Data columns (total 18 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   id                                     27901 non-null  int64  
 1   Gender                                 27901 non-null  object 
 2   Age                                    27901 non-null  float64
 3   City                                   27901 non-null  object 
 4   Profession                             27901 non-null  object 
 5   Academic Pressure                      27901 non-null  float64
 6   Work Pressure                          27901 non-null  float64
 7   CGPA                                   27901 non-null  float64
 8   Study Satisfaction                     27901 non-null  float64
 9   Job Satisfaction                       27901 non-null  float64
 10  Sleep Duration                         27901 non-null  object 
 11  Di

**ANALISIS DATASET :**
18 kolom dan 27,901 baris

### **1. Data Nominal (Kategori tanpa urutan)**
- **Gender** → (`Male`, `Female`)  
- **City** → (`Visakhapatnam`, `Bangalore`, `Srinagar`, `Varanasi`, `Jaipur`, dll.)  
- **Profession** → (`Student`, dll.)  
- **Dietary Habits** → (`Healthy`, `Moderate`, dll.)  
- **Degree** → (`B.Pharm`, `BSc`, `BA`, `BCA`, `M.Tech`, dll.)  
- **Have you ever had suicidal thoughts ?** → (`Yes`, `No`)  
- **Family History of Mental Illness** → (`Yes`, `No`)  
- **Depression** → (`1` = Depresi, `0` = Tidak depresi)  

---

### **2. Data Ordinal (Kategori dengan tingkatan)**
- **Academic Pressure** → (Skala `1` sampai `5`, semakin tinggi semakin besar tekanannya)  
- **Work Pressure** → (Skala `1` sampai `5`)  
- **Study Satisfaction** → (Skala `1` sampai `5`, semakin tinggi semakin puas)  
- **Job Satisfaction** → (Skala `1` sampai `5`)  
- **Financial Stress** → (Skala `1` sampai `5`, semakin tinggi semakin stres)  

---

### **3. Data Rasio (Numerik dengan nol absolut)**
- **Age** → (`24`, `25`, `28`, `31`, `33`, dll.)  
- **CGPA** → (`5.59`, `5.90`, `7.03`, `8.13`, `8.97`, dll.)  
- **Work/Study Hours** → (`1.0`, `3.0`, `4.0`, `9.0`, dll.)  

---

### **4. Data Interval (Numerik tanpa nol absolut)**
- **Sleep Duration** → (`Less than 5 hours`, `5-6 hours`, `7-8 hours`)  



